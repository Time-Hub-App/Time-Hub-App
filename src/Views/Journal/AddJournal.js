import React, { useState } from 'react';
import { useHistory } from 'react-router';

export default function AddJournal() {
    const [journal, setJournal] = useState('');
    const history = useHistory();

    const handleSubmit = async (e) => {
        e.preventDefault();
        try { await }
    }
    return (
        <div>
            
        </div>
    )
}
