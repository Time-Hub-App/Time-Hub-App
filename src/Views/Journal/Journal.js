import './Journal.css';
import Header from '../../Components/Header/Header.js';
import Footer from '../../Components/Footer/Footer.js';
import EntryList from '../../Components/EntryList/EntryList';
import { useState, useEffect } from 'react';
import { createEntry, fetchEntries } from '../../Services/journalEntries';
import EntryForm from '../../Components/EntryForm/EntryForm';

export default function Journal({ user, setUser }) {
  const [entries, setEntries] = useState([]);
  const [emotion, setEmotion] = useState('');
  const [text, setText] = useState('');
  const [loading, setLoading] = useState(true);
  useEffect(() => {
    const fetchData = async () => {
      const resp = await fetchEntries();
      setEntries(resp);
      setLoading(false);
    };
    fetchData();
  }, []);
  const formHandler = async (e) => {
    e.preventDefault();
    await createEntry(text, emotion);
  };

  if (loading) return <div>One second... all of your entries are coming!</div>;
  return (
    <div>
      <Header user={user} setUser={setUser} userpage />
      <EntryForm
        emotion={emotion}
        setEmotion={setEmotion}
        text={text}
        setText={setText}
        formHandler={formHandler}
      />
      <EntryList entries={entries} />
      <Footer />
    </div>
  );
}
