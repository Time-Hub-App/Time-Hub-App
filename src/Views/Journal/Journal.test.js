import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Journal from './Journal';

it('should render a header', () => {
  const { container } = render(
    <MemoryRouter>
      <Journal />
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
});
