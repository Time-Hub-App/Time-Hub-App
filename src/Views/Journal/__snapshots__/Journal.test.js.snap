// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a header 1`] = `
<div>
  <div>
    One second... all of your entries are coming!
  </div>
</div>
`;
